[CI] MixedClusterClientYamlTestSuiteIT test {p0=search.vectors/42_knn_search_bbq_flat/Test knn search} failing

**Build Scans:**
- [elasticsearch-intake #43272 / 9.4.4_bwc-snapshots](https://gradle-enterprise.elastic.co/s/umz255ht6wqk4)

**Reproduction Line:**
```
./gradlew ":qa:mixed-cluster:v9.4.4#mixedClusterTest" -Dtests.class="org.elasticsearch.backwards.MixedClusterClientYamlTestSuiteIT" -Dtests.method="test {p0=search.vectors/42_knn_search_bbq_flat/Test knn search}" -Dtests.seed=CC1044E1725EBF7A -Dtests.bwc=true -Dtests.locale=te-IN -Dtests.timezone=Universal -Druntime.java=26
```

**Applicable branches:**
9.5

**Reproduces locally?:**
N/A

**Failure History:**
[See dashboard](https://es-delivery-stats.elastic.dev/app/dashboards#/view/dcec9e60-72ac-11ee-8f39-55975ded9e63?_g=(refreshInterval:(pause:!t,value:60000),time:(from:now-7d%2Fd,to:now))&_a=(controlGroupState:(initialChildControlState:('0c0c9cb8-ccd2-45c6-9b13-96bac4abc542':(dataViewId:fbbdc689-be23-4b3d-8057-aa402e9ed0c5,fieldName:task.keyword,order:0,selectedOptions:!(),title:'GradleTask',type:optionsListControl),'4e6ad9d6-6fdc-4fcc-bf1a-aa6ca79e0850':(dataViewId:fbbdc689-be23-4b3d-8057-aa402e9ed0c5,fieldName:className.keyword,order:1,selectedOptions:!(org.elasticsearch.backwards.MixedClusterClientYamlTestSuiteIT),title:'Suite',type:optionsListControl),'144933da-5c1b-4257-a969-7f43455a7901':(dataViewId:fbbdc689-be23-4b3d-8057-aa402e9ed0c5,fieldName:name.keyword,order:2,selectedOptions:!('test%20%7Bp0%3Dsearch.vectors%2F42_knn_search_bbq_flat%2FTest%20knn%20search%7D'),title:'Test',type:optionsListControl)))))

**Failure Message:**
```
java.lang.AssertionError: org.junit.internal.runners.model.MultipleFailureException: There were 2 errors:
  java.lang.AssertionError(field [hits.hits.1._id] does not match the expected value
Expected: "3"
     but: was "2")
  java.lang.AssertionError(Error executing section at [search.vectors/42_knn_search_bbq_flat:105]: element was a list with 2 elements, but [2] was out of bounds)
```

**Issue Reasons:**
- [9.5] 1 consecutive failures in test test {p0=search.vectors/42_knn_search_bbq_flat/Test knn search}
- [9.5] 1 consecutive failures in step 9.4.4_bwc-snapshots
- [9.5] 1 consecutive failures in pipeline elasticsearch-intake

**Note:**
This issue was created using new test triage automation. Please report issues or feedback to es-delivery.

--- a/muted-tests.yml
+++ b/muted-tests.yml
@@ -571,9 +571,6 @@ tests:
 - class: org.elasticsearch.xpack.esql.ccq.MultiClusterSpecStatsIT
   method: test {csv-spec:stats.sumWithOverflowGroupingRow}
   issue: https://github.com/elastic/elasticsearch/issues/153434
-- class: org.elasticsearch.backwards.MixedClusterClientYamlTestSuiteIT
-  method: test {p0=search.vectors/42_knn_search_bbq_flat/Test knn search}
-  issue: https://github.com/elastic/elasticsearch/issues/153476
 - class: org.elasticsearch.xpack.monitoring.MultiNodesStatsTests
   method: testMultipleNodes
   issue: https://github.com/elastic/elasticsearch/issues/153497

